src/main/java/codeu/model/store/basic/ConversationStore.java

```diff
@@ -57,31 +57,54 @@ public static ConversationStore getTestInstance(PersistentStorageAgent persisten
 
   /** The in-memory list of Conversations. */
   private List<Conversation> conversations;
-	
-	/** The in-memory activity feed conversation. */
-	private Conversation actFeedConversation;
+
+  /** The in-memory activity feed conversation. */
+  private Conversation actFeedConversation;
+
+  /** The in-memory list of restricted conversation names reserved for private conversations. */
+  private List<String> restrictedConversationNames;
 
   /** This class is a singleton, so its constructor is private. Call getInstance() instead. */
   private ConversationStore(PersistentStorageAgent persistentStorageAgent) {
     this.persistentStorageAgent = persistentStorageAgent;
     conversations = new ArrayList<>();
+    restrictedConversationNames = new ArrayList<>();
   }
 
 	/** Access the current set of conversations known to the application. */
   public List<Conversation> getAllConversations() {
     return conversations;
   }
-	
+
 	/** Access the activity feed conversation known to the application. */
   public Conversation getActFeedConversation() {
     return actFeedConversation;
   }
 
+  /** Access the list of restricted conversation names. */
+  public List<String> getRestrictedConversationNames() {
+      return restrictedConversationNames;
+  }
+
+  /** Checks if a name exists in the list of restricted conversation names. */
+  public boolean isRestrictedName(String name) {
+      for (String n: restrictedConversationNames) {
```

Comment:
Note this is O(N) lookup.. if this is the only thing you are using restrictedConversationNames for, why not define it as a HashSet instead of a List? Then you would have O(1) lookup.. or is  there another reason to keep an (ordered) list?